connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

```diff
@@ -711,16 +728,24 @@ private void expectSendRecordSyncFailure(Throwable error) throws InterruptedExce
         return expectSendRecordTaskCommitRecordSucceed(false, isRetry);
     }
 
+    private Capture<ProducerRecord<byte[], byte[]>> expectSendRecordProducerCallbackFail(Exception producerCallbackException) throws InterruptedException {
+        return expectSendRecord(false, false, false, producerCallbackException);
+    }
+
     private Capture<ProducerRecord<byte[], byte[]>> expectSendRecordTaskCommitRecordSucceed(boolean anyTimes, boolean isRetry) throws InterruptedException {
-        return expectSendRecord(anyTimes, isRetry, true);
+        return expectSendRecord(anyTimes, isRetry, true, null);
```

Comment:
The `expectSendRecord` method is only invoked by three wrapper methods, seems fairly low-cost to add a parameter to it and then modify the bodies of those three wrappers. The most reasonable alternative I can think of would be to add another wrapper that those wrappers invoke that then calls `expectSendRecords` with a final parameter of `null`, but that seems more complex and wouldn't significantly decrease the number of affected lines.

Reply:
It’s quite hard to read a function call that has a null parameters being passed in. In this context it’s easy because the change was just made. This is a nit.

Reply:
Fair enough, I've simplified to use just a boolean instead of a full-blown `Exception`. LMK how that looks.